pkg/virt-operator/resource/apply/rbac.go

```diff
@@ -234,9 +235,9 @@ func enforceAPIGroup(existing runtime.Object, required runtime.Object) {
 	}
 }
 
-func arePolicyRulesEqual(role1 runtime.Object, role2 runtime.Object) (equal bool) {
+func changeExistingByRequired(existing runtime.Object, required runtime.Object) (changed bool) {
```

Comment:
this function is almost 100 lines. Let's break break it apart.

Reply:
the verb `change` and `changed` also reads weird. is this modified, or updated? By the name I can't tell if this is a compare function and its telling me something is different or if it's an action function and something was updated.

Reply:
Sure, `modified` is clearer.

Reply:
Regarding breaking the function apart, the reason this function is long is because helper functions are defined within it (`changeExistingPolicyRulesByRequired`, `changeExistingSubjectsByRequired`, ...). These helper functions are there, as stated in the comment, for performance reasons since it's faster than a simple deep reflection comparison.

IMHO if a long function is organized in a readable way and defines helper functions for its context that's not a problem per se. I think the alternative, which is extracting out these helper functions to regular private functions, is less clear since these helper functions belong to a very specific context which is now gone. In the future we could end up having a lot more functions in `rbac.go` and these would not be obvious and would make a mess.

Does that make sense to you?